### ISSUE TITLE
Hung with perl code `use open qw(:`...

### ISSUE THREAD

**Issue by sshilovsky** [helix-editor/helix#4070]
### Summary

When typing in this certain code sequence, helix hungs.

### Reproduction Steps

 

I tried this:

1. `hx test.pl`

2. In insert mode, type: `use open qw(:`

the final character `:` isn't displayed, helix seems to hung. Ctrl+C doesn't help

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2022-10-02T18:14:45.514 helix_view::editor [ERROR] Failed to initialize the LSP for `source.perl` { LSP not defined }

```

</details>

### Platform

Linux

### Terminal Emulator

alacritty 0.10.1 ()

### Helix Version

helix 22.08.1 (5b5f1bd3)

**Comment by A-Walrus** [comment]
Was able to reproduce as well. It only happens at the end of the document when there is no newline after it.

**Comment by pascalkuthe** [comment]
Closed by  #7644
